File: jruby-lab/analysis.m
javaaddpath "/path/to/your/project"

tt = TemperatureTaker();
tt.start();

printf("Listening for temperature for 10 seconds\n");

v = [];
for i = 1:10,
    sleep(1);
    v(i) = tt.temperature();
end
tt.stop();

plot(v);

File: jruby-lab/Analysis.java
public class Analysis {
    public static void main(String[] args) throws java.io.IOException {
        TemperatureTaker tt = new TemperatureTaker();
        tt.start();

        System.out.println("Listening for temperature; press Enter to stop");
        System.in.read();

        tt.stop();

        System.out.println("Temperature is " + tt.temperature());
    }
}
